src/BrazeKit-dev.js

```diff
@@ -666,27 +666,31 @@ var constructor = function () {
     function primeBrazeWebPush() {
         // The following code block is based on Braze's best practice for implementing
         // their push primer.  We only modify it to include pushPrimer and register_inapp settings.
-        // https://www.braze.com/docs/developer_guide/platform_integration_guides/web/push_notifications/integration/#soft-push-prompts
+        // https://www.braze.com/docs/developer_guide/platform_integration_guides/web/push_notifications/soft_push_prompt
         braze.subscribeToInAppMessage(function (inAppMessage) {
             var shouldDisplay = true;
             var pushPrimer = false;
             if (inAppMessage instanceof braze.InAppMessage) {
-                // Read the key-value pair for msg-id
-                var msgId = inAppMessage.extras['msg-id'];
-
-                // If this is our push primer message
-                if (msgId == 'push-primer') {
+                // access the key-value pairs, defined as `extras`
+                const keyValuePairs = inAppMessage.extras || {};
+                // check the value of our key `msg-id` defined in the Braze dashboard
+                if (keyValuePairs['msg-id'] === 'push-primer') {
                     pushPrimer = true;
-                    // We don't want to display the soft push prompt to users on browsers that don't support push, or if the user
-                    // has already granted/blocked permission
+                    // We don't want to display the soft push prompt to users on browsers
+                    // that don't support push, or if the user has already granted/blocked permission
                     if (
-                        !braze.isPushSupported() ||
+                        braze.isPushSupported() === false ||
```

Comment:
I think it might be cleaner to define a helper method called `isPushSupported()` that checks all of these values.

Reply:
While I agree, this is copied and pasted from their docs, so I want to make as minimal changes as possible unless it deals with us adding certain flags, like our variables for `shouldDisplay` and `pushPrimer`